src/SignalR/server/StackExchangeRedis/test/DefaultHubMessageSerializerTests.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.SignalR.Internal;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Tests.Internal
+{
+    public class DefaultHubMessageSerializerTests
+    {
+        [Theory]
+        [MemberData(nameof(InvocationTestData))]
+        public void SerializeMessages(string testName)
+        {
+            var testData = _invocationTestData[testName];
+
+            var resolver = CreateHubProtocolResolver(new List<IHubProtocol> { new MessagePackHubProtocol(), new JsonHubProtocol() });
+            var protocolNames = testData.SupportedHubProtocols.ConvertAll(p => p.Name);
+            var serializer = new DefaultHubMessageSerializer(resolver, protocolNames, hubSupportedProtocols: null);
```

Comment:
Do we have any tests that verify hubSupportedProtocols overrule the globalSupportedProtocols?